resources/idlharness.js

```diff
@@ -322,6 +322,155 @@ IdlArray.prototype.recursively_get_implements = function(interface_name)
     return ret;
 };
 
+//@}
+IdlArray.prototype.get_inheritance_stack = function(idl_interface)
+//@{
+{
+    /**
+     * See https://heycam.github.io/webidl/#create-an-inheritance-stack
+     *
+     * Returns an array of IdlInterface objects which contains idl_interface
+     * and all of its inherited interfaces.
+     *
+     * So given:
+     *
+     *   A : B {};
+     *   B : C {};
+     *   C {};
+     *
+     * then get_inheritance_stack(A) should return [A, B, C],
+     * and get_inheritance_stack(B) should return [B, C].
+     *
+     * Note: as dictionary inheritance is expressed identically by the AST,
+     * this works just as well for getting a stack of inherited dictionaries. 
+     */
+
+    var stack = [];
+    stack.push(idl_interface);
+    while (idl_interface.base) {
+        var base = this.members[idl_interface.base];
+        if (!base) {
+            throw new Error(idl_interface.type + " " + idl_interface.base + " not found (inherited by " + idl_interface.name + ")");
```

Comment:
I think it is by this: https://github.com/w3c/web-platform-tests/pull/6246/files#diff-061d4b3be7a8a8e16e9b15230b603d47R25